### ISSUE TITLE
CI Test failing after pyarrow release to v1.0.0

### ISSUE THREAD

**Issue by JimCircadian** [dask/dask#6459]
**What happened**:

All CI tests failing during to_parquet calls, reliant on apache arrow. Digging into the error and the timing it appears to correlate to the release of v1.0.0 over of v0.17.1 which is causing the CI tests to fail with the new version. The version specification currently sets only a lower bound >=0.14.0

**What you expected to happen**:

Tests run without trouble for unrelated development (I was doing doc development)

Raising as an issue just as it seems likely studying the trace that the tests need to be modified, but I'll not pick up doing that just for the moment so thought it best to share the experience, especially since it seems to be causing CI tests to fail for PRs! 

**Minimal Complete Verifiable Example**:

Simply install a fresh environment based on continuous_integration/environment-latest.yaml and run `pytest`

**Anything else we need to know?**:

Logs from both github CI and local machine attached at the bottom. Adding clause for <1.0.0 to the pyarrow dependency solved the issue:

```
@0.17.1 - 7283 passed, 422 skipped, 111 xfailed, 8 xpassed, 8121 warnings in 617.20s (0:10:17) 
@1.0.0 - 2 failed, 7281 passed, 422 skipped, 113 xfailed, 6 xpassed, 8138 warnings in 656.31s (0:10:56)
```

**Environment**:

As per environment-latest.yml

[pytest.pyarrow.log](https://github.com/dask/dask/files/4978251/pytest.pyarrow.log)

**Comment by gforsyth** [comment]
Thanks for raising this @JimCircadian -- definitely an issue.  Trying to figure out if we want to band-aid it in the interim or just fix it outright.

**Comment by jsignell** [comment]
This should be solved by #6346. Closing.

**Comment by jorisvandenbossche** [comment]
Note that the actual failures/changes are discussed in https://github.com/dask/dask/issues/6243
